appinventor/blocklyeditor/src/blocks/utilities.js

```diff
@@ -27,33 +42,70 @@ Blockly.Blocks.Utilities.InstantInTime = function () { return 'InstantInTime'; }
 // The Yail type 'any' is repsented by Javascript null, to match
 // Blockly's convention
 Blockly.Blocks.Utilities.YailTypeToBlocklyTypeMap = {
-  'number':{input:"Number",output:["Number","String", "Key"]},
-  'text':{input:"String",output:["Number","String", "Key"]},
-  'boolean':{input:"Boolean",output:["Boolean","String"]},
-  'list':{input:"Array",output:["Array","String"]},
-  'component':{input:"COMPONENT",output:["COMPONENT", "Key"]},
-  'InstantInTime':{input:Blockly.Blocks.Utilities.InstantInTime,output:Blockly.Blocks.Utilities.InstantInTime},
-  'any':{input:null,output:null},
-  'dictionary':{input:"Dictionary",output:["Dictionary", "String", "Array"]},
-  'pair':{input:"Pair",output:["Pair", "String", "Array"]},
-  'key':{input:"Key",output:["String", "Key"]}
-  //add  more types here
+  'number': {
+    input: ['Number'],
+    output: ['Number', 'String', 'Key']
+  },
+  'text': {
+    input: ['String'],
+    output: ['Number', 'String', 'Key']
+  },
+  'boolean': {
+    input: ['Boolean'],
+    output: ['Boolean', 'String']
+  },
+  'list': {
+    input: ['Array'],
+    output: ['Array', 'String']
+  },
+  'component': {
+    input: ['COMPONENT'],
+    output: ['COMPONENT', 'Key']
+  },
+  'InstantInTime': {
+    input: ['InstantInTime', Blockly.Blocks.Utilities.InstantInTime],
+    output: ['InstantInTime', Blockly.Blocks.Utilities.InstantInTime],
+  },
+  'any': {
+    input: null,
+    output: null
+  },
+  'dictionary': {
+    input: ['Dictionary'],
+    output: ['Dictionary', 'String', 'Array']
+  },
+  'pair': {
+    input: ['Pair'],
+    output: ['Pair', 'String', 'Array']
+  },
+  'key': {
+    input: ['Key'],
+    output: ['String', 'Key']
+  }
 };
 
-Blockly.Blocks.Utilities.OUTPUT = 1;
-Blockly.Blocks.Utilities.INPUT = 0;
+Blockly.Blocks.Utilities.OUTPUT = 'output';
```

Comment:
If we are going to look up by strings, we should quote the keys in the above dictionaries so that if we decide to move to advanced compilation the obfuscation doesn't take us by surprise.